src/definers.lisp

```diff
@@ -221,13 +221,14 @@
 (defun construct-function-definer-form (function name accessor)
   (let ((g!name (gensym "name")))
     (multiple-value-bind (arglist rest-argument normal-args k/o-args)
-        (normalize-arglist (if (symbolp function)
-                               (arglist function)
-                               (if (and (listp function)
-                                        (eq (first function)
-                                            'lambda))
-                                   (second function)
-                                   (error "Malformed function name"))))
+        (normalize-arglist (cond
+                             ((symbolp function)
+                              (t:arglist function))
+                             ((and (listp function)
+                                   (eq (first function)
+                                       'lambda))
+                              (second function))
+                             (t (error "Malformed function name"))))
```

Comment:
More descriptive error message would be welcome. Something like `(error "Malformed function name ~S while building the definer for ~S" function name)` perhaps?